pages/api/ccr.ts

```diff
@@ -0,0 +1,101 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { rankClient } from '../../services/elasticsearch'
+import { getSearchTemplates } from '../../services/search-templates'
+
+function getNamespace(indexName: string) {
+  return indexName.split('-')[0]
+}
+type State = {
+  index: string
+  state: 'delete' | 'exists' | 'follow'
+}
+export default async (
+  req: NextApiRequest,
+  res: NextApiResponse
+): Promise<void> => {
+  const auth = process.env.CCR_MANAGE_API_KEY
+  const authHeader = req.headers.authorization
+
+  if (!auth || !authHeader || auth !== process.env.CCR_MANAGE_API_KEY) {
+    res.statusCode = 401
+    res.setHeader('Content-Type', 'application/json')
+    res.end(
+      JSON.stringify({
+        statusCode: 401,
+        message: 'unauthorised',
+      })
+    )
+    return
+  }
+
+  const searchTemplates = await getSearchTemplates('prod')
+  const { body: allIndices } = await rankClient.indices.get({ index: '_all' })
+
+  const state: State[] = (
+    await Promise.all(
+      searchTemplates.templates.map(async (template) => {
+        const namespace = getNamespace(template.index)
+        const ccrIndexName = `ccr--${template.index}`
+        // we're only working with works for now, but should extend to images
```

Comment:
is there anything stopping us from including images now?

Reply:
The cluster size mainly. I'd like to get this working with works, and them try images, as that's a resources issue that will have it's own issues.